swingset unhandled promise lint fixes

I ran the new `scripts/lint-with-types.js` tool on swingset today, and here's a list of all the Promises that it thinks are not being fully handled:

```
node ➜ ~/agoric-sdk/packages/SwingSet $ ../../scripts/lint-with-types.sh
yarn run v1.22.17
$ yarn lint:types&&yarn lint:eslint
$ tsc --maxNodeModuleJsDepth 3 -p jsconfig.json
$ eslint .

/home/node/agoric-sdk/packages/SwingSet/misc-tools/measure-metering/measurement-target.js
  68:7  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

/home/node/agoric-sdk/packages/SwingSet/src/kernel/kernel.js
  266:7  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

/home/node/agoric-sdk/packages/SwingSet/src/kernel/vat-loader/manager-nodeworker.js
  67:7  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

/home/node/agoric-sdk/packages/SwingSet/src/liveslots/watchedPromises.js
   83:9   warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises
  117:11  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises
  173:5   warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

/home/node/agoric-sdk/packages/SwingSet/src/vats/network/network.js
  335:11  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises
  511:3   warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

/home/node/agoric-sdk/packages/SwingSet/src/vats/plugin-manager.js
  157:11  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises
  184:11  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises
  208:7   warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

/home/node/agoric-sdk/packages/SwingSet/src/vats/vattp/vat-vattp.js
  131:9  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

/home/node/agoric-sdk/packages/SwingSet/test/basedir-promises-2/bootstrap.js
  14:9  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises
  29:9  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

/home/node/agoric-sdk/packages/SwingSet/test/basedir-promises-3/vat-right.js
  12:7  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

/home/node/agoric-sdk/packages/SwingSet/test/basedir-promises/bootstrap.js
  13:9  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises
  14:9  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises
  40:9  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

/home/node/agoric-sdk/packages/SwingSet/test/device-plugin/test-device.js
  81:3  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises
  97:3  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

/home/node/agoric-sdk/packages/SwingSet/test/gc-device-transfer/bootstrap-gc.js
  19:7  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

/home/node/agoric-sdk/packages/SwingSet/test/test-liveslots.js
  222:9  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

/home/node/agoric-sdk/packages/SwingSet/test/test-network.js
   91:9  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises
  182:9  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

/home/node/agoric-sdk/packages/SwingSet/test/test-queue-priority.js
  10:5  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

/home/node/agoric-sdk/packages/SwingSet/test/workers/vat-target.js
  60:5  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

/home/node/agoric-sdk/packages/SwingSet/tools/vat.js
  54:7  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises
  58:7  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises
  63:1  warning  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises

✖ 34 problems (0 errors, 34 warnings)
```

We should fix these.

--- a/packages/SwingSet/misc-tools/measure-metering/measurement-target.js
+++ b/packages/SwingSet/misc-tools/measure-metering/measurement-target.js
@@ -65,7 +65,7 @@ export function buildRootObject(vatPowers) {
       purse.deposit(payment);
     },
     getUpdateSince() {
-      notifier.getUpdateSince();
+      return notifier.getUpdateSince();
     },
   });
 }

--- a/packages/SwingSet/src/kernel/kernel.js
+++ b/packages/SwingSet/src/kernel/kernel.js
@@ -264,7 +264,7 @@ export default function buildKernel(
 
       // ISSUE: terminate stuff in its own crank like creation?
       // eslint-disable-next-line no-use-before-define
-      vatWarehouse.vatWasTerminated(vatID);
+      void vatWarehouse.vatWasTerminated(vatID);
     }
     if (critical) {
       // The following error construction is a bit awkward, but (1) it saves us

--- a/packages/SwingSet/src/kernel/vat-loader/manager-nodeworker.js
+++ b/packages/SwingSet/src/kernel/vat-loader/manager-nodeworker.js
@@ -64,7 +64,7 @@ export function makeNodeWorkerVatManagerFactory(tools) {
 
     function sendToWorker(msg) {
       assert(msg instanceof Array);
-      workerP.then(worker => worker.postMessage(msg));
+      void workerP.then(worker => worker.postMessage(msg));
     }
 
     const { promise: dispatchReadyP, resolve: dispatchIsReady } =

--- a/packages/SwingSet/src/liveslots/watchedPromises.js
+++ b/packages/SwingSet/src/liveslots/watchedPromises.js
@@ -80,7 +80,7 @@ export function makeWatchedPromiseManager(
       promiseRegistrations.delete(vpid);
       for (const watch of watches) {
         const [watcher, ...args] = watch;
-        Promise.resolve().then(() => {
+        void Promise.resolve().then(() => {
           if (wasFulfilled) {
             if (watcher.onFulfilled) {
               watcher.onFulfilled(value, ...args);
@@ -114,7 +114,7 @@ export function makeWatchedPromiseManager(
         watchedPromiseTable.delete(vpid);
         for (const watch of watches) {
           const [watcher, ...args] = watch;
-          Promise.resolve().then(() => {
+          void Promise.resolve().then(() => {
             if (watcher.onRejected) {
               watcher.onRejected(VatUpgradedMessage, ...args);
             } else {
@@ -170,7 +170,7 @@ export function makeWatchedPromiseManager(
     // TODO: remove mapping upon resolution
     // TODO: track watched but non-exported promises, add during prepareShutdownRejections
     //  maybe check importedVPIDs here and add to table if !has
-    Promise.resolve().then(() => {
+    void Promise.resolve().then(() => {
       const watcherVref = convertValToSlot(watcher);
       assert(watcherVref, 'invalid watcher');
       const { virtual } = parseVatSlot(watcherVref);

--- a/packages/SwingSet/src/vats/network/network.js
+++ b/packages/SwingSet/src/vats/network/network.js
@@ -332,7 +332,7 @@ export function makeNetworkProtocol(protocolHandler) {
         // eslint-disable-next-line no-use-before-define
         const conn = await protocolImpl.outbound(port, dst, connectionHandler);
         if (revoked) {
-          E(conn).close();
+          void E(conn).close();
         } else {
           openConnections.add(conn);
         }
@@ -508,7 +508,7 @@ export function makeNetworkProtocol(protocolHandler) {
   });
 
   // Wire up the local protocol to the handler.
-  E(protocolHandler).onCreate(protocolImpl, protocolHandler);
+  void E(protocolHandler).onCreate(protocolImpl, protocolHandler);
 
   // Return the user-facing protocol.
   return Far('binder', { bind });

--- a/packages/SwingSet/src/vats/plugin-manager.js
+++ b/packages/SwingSet/src/vats/plugin-manager.js
@@ -154,7 +154,7 @@ export function makePluginManager(pluginDevice, { D, ...vatPowers }) {
           pluginRootPK = makePromiseKit();
 
           // Tell our clients we are resetting.
-          E(resetter).onReset(pluginRootPK.promise.then(_ => true));
+          void E(resetter).onReset(pluginRootPK.promise.then(_ => true));
 
           // Attempt to restart the protocol using the same device connection.
           connect();
@@ -181,7 +181,7 @@ export function makePluginManager(pluginDevice, { D, ...vatPowers }) {
         makeStableForwarder(walker = DEFAULT_WALKER) {
           let pr;
           // eslint-disable-next-line no-new
-          new HandledPromise((_resolve, _reject, resolveWithPresence) => {
+          void new HandledPromise((_resolve, _reject, resolveWithPresence) => {
             // Use Remotable rather than Far to make a remote from a presence
             pr = Remotable(
               'Alleged: stableForwarder',
@@ -205,7 +205,7 @@ export function makePluginManager(pluginDevice, { D, ...vatPowers }) {
       });
 
       // Declare the first reset.
-      E(resetter).onReset(Promise.resolve(false));
+      void E(resetter).onReset(Promise.resolve(false));
 
       // Start the first connection.
       connect();

--- a/packages/SwingSet/src/vats/vattp/vat-vattp.js
+++ b/packages/SwingSet/src/vats/vattp/vat-vattp.js
@@ -128,7 +128,7 @@ export function buildRootObject(vatPowers) {
         console.warn(`deleting connection is not fully supported in comms`);
       },
       infoMessage(...args) {
-        E(cons).log('VatTP connection info:', ...args);
+        void E(cons).log('VatTP connection info:', ...args);
       },
     });
 

--- a/packages/SwingSet/test/basedir-promises-2/bootstrap.js
+++ b/packages/SwingSet/test/basedir-promises-2/bootstrap.js
@@ -11,7 +11,7 @@ export function buildRootObject(vatPowers, vatParameters) {
       const mode = argv[0];
       if (mode === 'harden-promise-1') {
         const { promise: p1 } = makePromiseKit();
-        harden(p1);
+        void harden(p1);
         const allP = [];
         // in bug #95, this first call returns a (correctly) frozen Promise,
         // but for the wrong reasons
@@ -26,7 +26,7 @@ export function buildRootObject(vatPowers, vatParameters) {
         // const p4 = vats.left!checkHarden(p1);
         // log(`p4 frozen ${Object.isFrozen(p4)}`);
         // allP.push(p4);
-        Promise.all(allP).then(_ => {
+        return Promise.all(allP).then(_ => {
           log(`b.harden-promise-1.finish`);
         });
       } else {

--- a/packages/SwingSet/test/basedir-promises-3/vat-right.js
+++ b/packages/SwingSet/test/basedir-promises-3/vat-right.js
@@ -9,7 +9,7 @@ export function buildRootObject() {
       // arg should be a Promise that promptly resolves to 4
       const argP = Promise.resolve(arg);
       const wasP = argP === arg;
-      argP.then(newArg => pk4.resolve([wasP, newArg]));
+      return argP.then(newArg => pk4.resolve([wasP, newArg]));
     },
   });
 

--- a/packages/SwingSet/test/basedir-promises/bootstrap.js
+++ b/packages/SwingSet/test/basedir-promises/bootstrap.js
@@ -10,8 +10,8 @@ export function buildRootObject(vatPowers, vatParameters) {
     bootstrap(vats) {
       const mode = vatParameters.argv[0];
       if (mode === 'flush') {
-        Promise.resolve('then1').then(log);
-        Promise.resolve('then2').then(log);
+        void Promise.resolve('then1').then(log);
+        void Promise.resolve('then2').then(log);
       } else if (mode === 'e-then') {
         E(vats.left)
           .callRight(1, vats.right)
@@ -37,7 +37,7 @@ export function buildRootObject(vatPowers, vatParameters) {
           },
         });
         const p1 = E(t1).foo(1);
-        p1.then(x => log(`b.resolved ${x}`));
+        void p1.then(x => log(`b.resolved ${x}`));
         log(`b.local1.finish`);
       } else if (mode === 'local2') {
         const t1 = Far('t1', {

--- a/packages/SwingSet/test/device-plugin/test-device.js
+++ b/packages/SwingSet/test/device-plugin/test-device.js
@@ -78,7 +78,7 @@ test.serial('plugin first time', async t => {
     t,
   );
 
-  queueThunkForKernel(() => bridge.deliverInbound('pingpong'));
+  void queueThunkForKernel(() => bridge.deliverInbound('pingpong'));
   await cycle();
 
   t.deepEqual(dump().log, [
@@ -94,7 +94,7 @@ test.serial('plugin after restart', async t => {
     await setupVatController(t);
 
   plugin.reset();
-  queueThunkForKernel(() => bridge.deliverInbound('pingpong'));
+  void queueThunkForKernel(() => bridge.deliverInbound('pingpong'));
   await cycle();
 
   t.deepEqual(dump().log, [

--- a/packages/SwingSet/test/gc-device-transfer/bootstrap-gc.js
+++ b/packages/SwingSet/test/gc-device-transfer/bootstrap-gc.js
@@ -16,7 +16,7 @@ export function buildRootObject(vatPowers) {
     async bootstrap(vats, devices) {
       pin.push(vats); // don't drop vat-right, our test needs it later
       // export amy to the device, and send to vat-left, then forget her
-      sendAmy(D, devices.stash_device, testLog, vats.left);
+      void sendAmy(D, devices.stash_device, testLog, vats.left);
       // tell the vat to retrieve amy from the device
       await E(vats.right).acceptDevice(devices.stash_device);
     },

--- a/packages/SwingSet/test/test-liveslots.js
+++ b/packages/SwingSet/test/test-liveslots.js
@@ -219,7 +219,7 @@ async function doOutboundPromise(t, mode) {
         // Queue up a call that includes the promise again. This will run
         // *after* the promise has been resolved. Our current implementation
         // will use the same promise identifier.
-        Promise.resolve().then(() => E(target).two(p));
+        void Promise.resolve().then(() => E(target).two(p));
       },
     });
   }

--- a/packages/SwingSet/test/test-network.js
+++ b/packages/SwingSet/test/test-network.js
@@ -88,7 +88,7 @@ test('handled protocol', async t => {
         const ack = await connection.send('ping');
         // log(ack);
         t.is(`${ack}`, 'ping', 'received pong');
-        connection.close();
+        void connection.close();
       },
       async onClose(_connection, reason) {
         t.is(reason, undefined, 'no close reason');
@@ -179,7 +179,7 @@ test('protocol connection listen', async t => {
         if (connectionHandler.onOpen) {
           await connectionHandler.onOpen(connection, localAddr, remoteAddr, c);
         }
-        connection.send('ping');
+        void connection.send('ping');
       },
     }),
   );

--- a/packages/SwingSet/test/test-queue-priority.js
+++ b/packages/SwingSet/test/test-queue-priority.js
@@ -7,7 +7,7 @@ test('Promise queue should be higher priority than IO/timer queue', async t => {
   setImmediate(() => log.push(1));
   setImmediate(() => {
     log.push(2);
-    Promise.resolve().then(() => log.push(4));
+    void Promise.resolve().then(() => log.push(4));
     log.push(3);
   });
   setImmediate(() => log.push(5));

--- a/packages/SwingSet/test/workers/vat-target.js
+++ b/packages/SwingSet/test/workers/vat-target.js
@@ -57,7 +57,7 @@ export function buildRootObject(vatPowers, vatParameters) {
   //   syscall.reject(pC, Error('oops'))
   //   syscall.fulfillToData(rp3, 'rp3 good')
   function one() {
-    Promise.reject(Error('one unhandled rejection'));
+    void Promise.reject(Error('one unhandled rejection'));
     precB.resolve(callbackObj); // syscall.fulfillToPresence
     precC.reject(Error('oops')); // syscall.reject
     return 'rp3 good';

--- a/packages/SwingSet/tools/vat.js
+++ b/packages/SwingSet/tools/vat.js
@@ -51,13 +51,13 @@ async function main() {
     r.context.dump2 = () => controller.dump();
     r.context.run = () => {
       console.log('run!');
-      controller.run();
+      return controller.run();
     };
     r.context.step = () => {
       console.log('step!');
-      controller.step();
+      return controller.step();
     };
   }
 }
 
-main();
+void main();

